### ISSUE TITLE
Omnicomplete writes to disk when selecting suggestion - plays badly with YCM

### ISSUE THREAD

**Issue by maackle** [FrigoEU/psc-ide-vim#91]
I'm not sure if this is intended or not, and I'm new to vim so I'm not sure if this is a common or even unavoidable thing, but: when omnicompleting, the buffer gets written to disk whenever selecting an omnicomplete suggestion. This is bad in its own right, but it also plays really badly with [YouCompleteMe](https://github.com/Valloric/YouCompleteMe), since in that case, it writes to disk *every time a character is typed* which caused an autocomplete suggestion.

I'd like to try to fix this but I would first like to know the rationale for writing to disk in the first place with omnicomplete, so I can get some direction.

**Comment by coot** [comment]
That's a valid issue. It should not happen.  We used to save when importing identifiers, but now it happens through vim rather than letting `purs ide` change files in place.  I cannot reproduce it.

**Comment by maackle** [comment]
That seems to be related, and still an issue for me. Here's a few lines from the message log (`g:psc_ide_log_level == 3`) where the writing happens. This was just from using autocomplete, not even calling importIdentifier.

```
purescript#ide#callSync: command: {"params":{"options":{"groupReexports":true},"currentModule":"App.View.Homepage","matcher":{"params":{"search":"ul"},"matche
r":"flex"},"filters":[{"filter":"prefix","params":{"search":"ul"}}]},"command":"complete"}
purescript#ide#callSync: trying to reach server again
s:callFn: response: <MANY LINES OF DATA>
...
PSCIDEimportIdentifier
ident: ul
"frontend/src/App/View/Homepage.purs" 73L, 2529C written
```

Rather than wonder if importIdentifier should be called during autocomplete, I'll ask about the fact that importIdentifier makes a call to `purescript#ide#utils#update()` [here](https://github.com/FrigoEU/psc-ide-vim/blob/master/autoload/purescript/ide/import.vim#L274), which calls `update` which should write the buffer, right?

**Comment by coot** [comment]
Yes, but we might not need this call any more.  `importIdentifier` is called on `CompleteDone` event and you can disable it with `let ide_import_on_comletion = v:false`.

There are various other places where we save the buffer.  For example when `:Prebuild` runs.  But that could also be avoided by saving the file to a temporary file and sending that file to recompile.

**Comment by ChrisPenner** [comment]
I'm suffering from this issue too; kinda making psc-ide unusable tbh.

Setting `let g:ide_import_on_completion = v:false` helps a lot 👍 ; I'd still love to see a more permanent fix though!

EDIT: Looks like some sort of automation is also automatically saving when I exit insert mode?? This whole hijacking my editor thing doesn't really work with my workflow. I'd much rather set up any autocmds that I need myself (though I understand that good defaults are nice)

**Comment by dakom** [comment]
I'm also having major trouble with this and YCM ... was driving me nuts for hours since the bug was something like, editing _non-purescript files_ (like typescript) would trigger a webpack reload when various keys are pressed in insert mode. Yeah, really... 

Was going to open a separate issue but it sounds highly related to this
